### PR TITLE
feat: bump to 0.3.1, remove from global scannerMode and onPremCompatibilityVersion

### PR DESCRIPTION
**Pull request by ProvoK**
## Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [ ] Title of the PR starts with type and scope, (e.g. feat(agent,node-analyzer,sysdig-deploy):)
- [ ] Chart Version bumped for the respective charts
- [ ] Variables are documented in the README.md (or README.tpl in some charts)
- [ ] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [ ] All test files are added in the tests folder of their respective chart and have a "_test" suffix

Check Contribution guidelines in README.md for more insight.

### CODE REVIEW

File: charts/cluster-scanner/templates/_helpers.tpl [hunk 1]
```diff
@@ -225,7 +223,7 @@ Version tags must be semver2-compatible otherwise it fails.
 */}}
 {{- define "cluster-scanner.checkVersionCompatibility" -}}
 {{- $version := semver .Tag -}}
-{{- if and .Values.global (hasKey (default .Values.global dict) "onPremCompatibilityVersion") (eq .Values.global.onPremCompatibilityVersion "6.2") -}}
+{{- if and .Values.global (hasKey (default .Values dict) "onPremCompatibilityVersion") (eq .Values.onPremCompatibilityVersion "6.2") -}}
```
Comment (michele-mangili) [inline charts/cluster-scanner/templates/_helpers.tpl]:
```suggestion
{{- if and (hasKey (default .Values dict) "onPremCompatibilityVersion") (eq .Values.onPremCompatibilityVersion "6.2") -}}
```

File: charts/cluster-scanner/templates/configmap.yaml [hunk 2]
```diff
@@ -13,9 +13,6 @@ data:
   eve_enabled: {{ .Values.eveEnabled | quote}}
   eve_integration_enabled: {{ .Values.eveIntegrationEnabled | quote }}
   logging_level: {{ .Values.global.loggingLevel }}
-{{- if eq .Values.global.scannerMode "local" }}
-  local_registry_secrets: {{ include "cluster-scanner.runtimeStatusIntegrator.localCluster.localSecrets" . }}
-{{- end }}
```
Comment (michele-mangili) [inline charts/cluster-scanner/templates/configmap.yaml:18]: I am not sure why we removed this. In the `deployment.yaml` this config map still exists. :thinking:

Comment (ProvoK) [reply charts/cluster-scanner/templates/configmap.yaml:18]: I've moved inside `cluster-scanner.modeConfig` helper func, which is used in the next line!

I though it was a good refactor so to have less places where `.Values.scannerMode` is checked

Comment (michele-mangili) [reply charts/cluster-scanner/templates/configmap.yaml:18]: oh you're right, I missed that `cluster-scanner.modeConfig` was included in the next line. Got it, thanks!

**Review by michele-mangili — APPROVED** [review]

**Pull request merged**